### PR TITLE
Update all the things

### PR DESCRIPTION
**Pull request by eduardb**
#### Problem/Goal

The dependencies of this project are outdated, let's update them! 

#### Solution

- Android Tools Gradle plugin updated from 1.5.0 to 2.3.1
- Updated Bintray Release plugin from 0.3.4 to the latest version, 0.4.0
- Updated Gradle Wrapper version from 2.2.3 to 3.3 (this was the latest version with which the Bintray Release plugin still works)
- Target and compile SDK updated from 23 to 25
- Build tools updated from 22.0.1 to 25.0.2
- Support libs updated from 23.2.1 to 25.3.1
- Using `support-core-ui` instead of `support-v4` to avoid pulling in unnecessary dependencies like `media-compat`, as seen in the screenshots below

##### Test(s) added 

Nothing to test

##### Screenshots

| Before | After |
| ------ | ----- |
| screenshot 2017-04-11 13 16 45 | screenshot 2017-04-11 12 05 22 |

### CODE REVIEW

File: library/build.gradle [hunk 1]
```diff
@@ -2,17 +2,17 @@ apply plugin: 'com.android.library'
 apply plugin: 'bintray-release'
 
 dependencies {
-    compile 'com.android.support:support-v4:23.2.1'
+    compile 'com.android.support:support-core-ui:25.3.1'
```
Comment (ataulm) [inline library/build.gradle:5]: 👀 is this where ViewPager lives now?

File: library/build.gradle [hunk 2]
```diff
@@ -2,17 +2,17 @@ apply plugin: 'com.android.library'
...
 }
 
 android {
     defaultConfig {
         minSdkVersion 16
-        targetSdkVersion 23
+        targetSdkVersion 25
```
Comment (ataulm) [inline library/build.gradle]: can we delete this line - don't need to specify the target in the library module

File: library/build.gradle (hunk 1 above)
Comment (eduardb) [reply library/build.gradle:5]: Yup, I checked before 😂

**Review by ataulm — APPROVED** [review]
:shipit:

File: library/build.gradle (hunk 2 above)
Comment (eduardb) [reply library/build.gradle]: Good point, done in af3932e

**Pull request merged**